File: packages/sdk/src/resources/workflow-runs.ts
import type { WorkflowRunId } from "@osva/contracts";
import {
  createWorkflowRunRequestSchema,
  workflowRunResourceSchema,
} from "@osva/contracts/schemas";
import type { z } from "zod";

import type { OsvaHttpClient } from "../http-client.js";

type CreateWorkflowRunRequest = z.infer<typeof createWorkflowRunRequestSchema>;
type WorkflowRunResource = z.infer<typeof workflowRunResourceSchema>;

export class WorkflowRunsResource {
  constructor(
    private readonly client: OsvaHttpClient,
    private readonly workspaceId: string,
  ) {}

  create(
    input: Omit<CreateWorkflowRunRequest, "workspaceId">,
  ): Promise<WorkflowRunResource> {
    return this.client.request({
      method: "POST",
      path: "/v1/workflow-runs",
      body: {
        ...input,
        workspaceId: this.workspaceId,
      },
    });
  }

  get(workflowRunId: WorkflowRunId): Promise<WorkflowRunResource> {
    return this.client.request({
      method: "GET",
      path: `/v1/workflow-runs/${encodeURIComponent(workflowRunId)}`,
    });
  }
}

File: sdks/python/src/osva/resources/workflow_runs.py
from __future__ import annotations

from typing import Any

from osva.http_client import HTTPClient


class WorkflowRunsResource:
    def __init__(self, client: HTTPClient) -> None:
        self._client = client

    def create(self, payload: dict[str, Any]) -> dict[str, Any]:
        return self._client.request("POST", "/v1/workflow-runs", body=payload)

    def get(self, workflow_run_id: str) -> dict[str, Any]:
        return self._client.request("GET", f"/v1/workflow-runs/{workflow_run_id}")
